src/test/java/org/I0Itec/zkclient/SaslAuthenticatedTest.java

```diff
@@ -150,11 +151,35 @@ public void testAuthFailure_AllowFailedSasl() throws IOException {
             bootstrapWithAuthFailure();
             fail("Expected to fail!");
         } catch (ZkException e) {
-            assertThat(e).isInstanceOf(ZkTimeoutException.class);
+            assertThat(e).isInstanceOf(ZkAuthFailedException.class);
         } finally {
             System.clearProperty(ZK_ALLOW_FAILED_SASL);
         }
     }
+    
+    /**
+     * Tests that an unauthenticated client can execute operations.
+     * 
+     * @throws IOException
+     */
+    @Test
+    public void testUnauthenticatedClient() throws IOException {
+        ZkClient unauthed = null;
+
+        try {
+            bootstrap();
+            System.clearProperty(ZkClient.JAVA_LOGIN_CONFIG_PARAM);
+            System.setProperty("zookeeper.sasl.client", "false");
+            unauthed = new ZkClient("localhost:" + _port, 6000);
+            unauthed.createPersistent("/test", new byte[0], Ids.OPEN_ACL_UNSAFE);
+        } catch (ZkException e) {
```

Comment:
Regarding this test... the unauthed.createPersistent() call is supposed to succeed, right ?